airflow/providers/apache/spark/hooks/spark_submit.py

```diff
@@ -98,10 +98,42 @@ class SparkSubmitHook(BaseHook, LoggingMixin):
     def get_ui_field_behaviour(cls) -> dict[str, Any]:
         """Return custom field behaviour."""
         return {
-            "hidden_fields": ["schema", "login", "password"],
+            "hidden_fields": ["schema", "login", "password", "extra"],
             "relabeling": {},
         }
 
+    @classmethod
+    def get_connection_form_widgets(cls) -> dict[str, Any]:
+        """Returns connection widgets to add to connection form."""
+        from flask_appbuilder.fieldwidgets import BS3TextFieldWidget
+        from flask_babel import lazy_gettext
+        from wtforms import StringField
+        from wtforms.validators import Optional, any_of
+
+        return {
+            "queue": StringField(
+                lazy_gettext("YARN queue"),
+                widget=BS3TextFieldWidget(),
+                description="Default YARN queue to use",
+                validators=[Optional()],
+            ),
+            "deploy-mode": StringField(
+                lazy_gettext("Deploy mode"),
+                widget=BS3TextFieldWidget(),
+                description="Must be client or cluster",
+                validators=[any_of(["client", "cluster"])],
+            ),
+            "spark-binary": StringField(
+                lazy_gettext("Spark binary"),
+                widget=BS3TextFieldWidget(),
+                description=f"Must be one of: {', '.join(ALLOWED_SPARK_BINARIES)}",
+                validators=[any_of(ALLOWED_SPARK_BINARIES)],
```

Comment:
WDYT about making these [`SelectField`](https://wtforms.readthedocs.io/en/2.3.x/fields/#wtforms.fields.SelectField)s instead?

Reply:
I've tried but it won't get rendered.
It looks like there is a restriction on which field we can use.
https://github.com/apache/airflow/blob/73d87945e9a78195278d0a4b495483062ddc9b35/airflow/providers_manager.py#L948

Reply:
@josh-fell 
I can add it into the `allowed_field_classes`, WDYT?
Or maybe I should do it in a different PR?

Reply:
This change in Airflow core, so it would not work in providers until it meet minimum Airflow version where this change would be applied, approx 1 year after commit to the core

Reply:
I see.
So for now I will keep it as it is :)

Reply:
That is might be a good idea extend allowed list by a separate PR, so one day it is possible to extend community providers connection UI.

Reply:
Agreed, separate PR, but there was a [particular reason](https://github.com/apache/airflow/issues/17193) why `SelectField` wasn't in the original list of allowed types it seems. The issue is 2 years old, but obviously we don't want to resurface an old bug. Just a heads up if/when making the change later. 

It would be very useful to have this form field type. (It's used in the Connection form itself already.)